src/main/java/io/neonbee/NeonBee.java

```diff
@@ -314,9 +317,34 @@ static Future<Vertx> newVertx(VertxOptions vertxOptions, NeonBeeOptions options)
         vertxOptions.getEventBusOptions().setPort(options.getClusterPort());
         Optional.ofNullable(getHostIp()).filter(Predicate.not(String::isEmpty))
                 .ifPresent(currentIp -> vertxOptions.getEventBusOptions().setHost(currentIp));
-        return Vertx.clusteredVertx(vertxOptions).onFailure(throwable -> {
-            LOGGER.error("Failed to start clustered Vert.x", throwable); // NOPMD slf4j
-        });
+
+        return applyEncryptionOptions(options, vertxOptions.getEventBusOptions())
+                .compose(v -> Vertx.clusteredVertx(vertxOptions)).onFailure(throwable -> {
+                    LOGGER.error("Failed to start clustered Vert.x", throwable); // NOPMD slf4j
+                });
+    }
+
+    @SuppressWarnings("PMD.EmptyIfStmt")
```

Comment:
I find the @SuppressWarnings("PMD.EmptyIfStmt") unnecessary. Why not simply return in each case a future?